src/libaktualizr-posix/asn1/asn1_message.cc

```diff
@@ -53,14 +54,16 @@ void SetString(OCTET_STRING_t* dest, const std::string& str) {
   OCTET_STRING_fromBuf(dest, str.c_str(), static_cast<int>(str.size()));
 }
 
-Asn1Message::Ptr Asn1Rpc(const Asn1Message::Ptr& tx, const struct sockaddr_storage& client) {
-  int socket_fd = socket(AF_INET6, SOCK_STREAM, 0);
+Asn1Message::Ptr Asn1Rpc(const Asn1Message::Ptr& tx, const struct sockaddr_in& server_sock_addr) {
+  int socket_fd = socket(AF_INET, SOCK_STREAM, 0);
```

Comment:
`AF_INET6` is no good?

Reply:
Benefit of ipv6 usage is questionable in this case (local network within a car).

Reply:
What are the drawbacks?

Reply:
Well, 
1) people get used to IPv4 addresses, so easier to edit a configuration file (::1 instead of  127.0.0.1 and IPv6 slightly longer
2) At the kernel level (IP stack implementation) slightly more work for packets processing 

We consider IPUptaneSecondary as a reference implementation, so I tend to make it simple, unless there is a requirement to support IPv6

Reply:
I actually have no strong feelings here. I'm fine with keeping it simple for now.

Reply:
+1 for ipv4. I can easily imagine ipv6 stack being disabled completely in an embedded system, as it generates additional neighbor discovery multicast traffic. While I'm totally in favor of using ipv6 for large scale networking, it looks like an extra burden for an isolated network in a vehicle.

Reply:
That might be not the case for IoT, though ;)

Reply:
I don't really buy the performance considerations if not tied to a specific platform/system architecture. Disabling v6 support also seems easier than adding it after the fact to a v4 only code base.

That said, I would have hoped that we wouldn't have to worry about either v4 or v6 by using dual stack sockets. If that's not the case and makes development harder, then ok for dropping it.

Reply:
It's worth keeping in mind this whole functionality is just meant to be a reference and demonstration; so it is perfectly acceptable to make a semi-arbitrary choice and just roll with it.

Reply:
Fair point, I forgot that it will actually support both AF_INET and AF_INET6. Then I'm for leaving it as it is :)

Reply:
@patrickvacek sure, sounds reasonable

Reply:
The previous version supported only AF_INET6 and it's used the 'discovery' mechanism to get IPv6 addresses. Since, we are going to remove the 'discovery' mechanism and input IP addresses in the config file then there is no need to stay with IPv6. 
I prefer to choose either IPv4 or IPv6, adding support of both will require additional effort, but I am  fine with either choices.